Fix ripple effect overflow in Tab bar

codebu5ter opened the pull request:
Before: 
![37422120-b48848e2-27e0-11e8-99ab-545b557f89d9](https://user-images.githubusercontent.com/16636569/37564516-8a12d818-2abd-11e8-9df0-0ddfa5f5cd4c.png)

After:
![screenshot_20180318-135710](https://user-images.githubusercontent.com/16636569/37564520-93d18494-2abd-11e8-80c6-90a027e01069.png)


Fixes #2065

borisyankov commented:
Cool. That is a nice progress. 👍 

I will give you a big-picture review first, and after you address these notes, we can go over it on code-level.

Test:  make sure the tests are passing. You can do these checks locally regularly to verify your code's correctness.

Code formatting: Have a look at `prettier`. We use it as automatic code formatting in our code editors. You can manually run it via `npm run prettier`.

Commit description: it is awesome you provided screenshots of before/after. Please, give one-two centence high-level description of what the fix consists of (something like 'A new wrapper component is created because React Navigation did this and this')

Lastly: go over the code again and see if you can implement this in fewer lines. I didn't go too much in detail through the code, so it is possible this particular fix requires that much code change, but maybe you can think of less changes for the same fix (smaller changes - less chance for regressions)

codebu5ter commented:
@borisyankov Thanks for the prompt review. 😃 
Having set up `prettier`, I've made a new commit which has passed both tests.
I've also updated the commit description to reflect what my commit is about.
Lastly, I've been trying to solve this issue for the past 2 days and I always got stuck while fetching the current theme of the app. The way I've implemented it is to just wrap the `TabNavigator` component in a class to access `theme` props, which are passed through by `MainScreenWithTabs.js`and help in dynamically changing the background color of the TabBar.
So, it's just the wrapping up of the component and the subsequent re-formatting of the code that has increased the number of lines changed when, in essence, the real code additions are much smaller.

jainkuniya commented on src/styles/tabs.js:
```diff
@@ -32,7 +40,7 @@ export default ({ tabBarComponent, tabBarPosition, showLabel, showIcon, tabWidth
     },
     style: {
-      backgroundColor: 'transparent',
+      backgroundColor: theme === 'night' ? '#212D3B' : 'white',
```
Try to reuse `#212D3B` from https://github.com/zulip/zulip-mobile/blob/bd915517f3f45d49d0e0174bc297409724331186/src/styles/themeDark.js#L4. So in later stage if we change, it will be easy & less error prone.

codebu5ter replied:
Done. 👍

jainkuniya commented on src/main/StreamTabs.js:
```diff
@@ -15,34 +15,45 @@ const styles = StyleSheet.create({
+  theme: string,
+};
+
+export default class DynamicTabBar extends Component<Props> {
```
Rename `DynamicTabBar` to more clearer name.

codebu5ter replied:
Renamed both classes to better reflect their positions.

codebu5ter commented:
@jainkuniya I took a look at the commit message guideline and I'll rebase all my commits to reflect the updated commit message once this PR is approved. 👍 

About making the code concise, the main issue stems from the fact that the `tabsOptions` method inside the `TabNavigator` is unable to access the `screenProps`. I tried a bunch of different methods, but this is the only one that worked (ie. wrapping TabNavigator in a class).
